src/indexBy.test.ts

```diff
@@ -6,17 +6,30 @@ const array = [
   { dir: 'right', code: 100 },
 ] as const;
 const expected = {
-  left: { dir: 'left', code: 97 },
-  right: { dir: 'right', code: 100 },
+  '97': { dir: 'left', code: 97 },
+  '100': { dir: 'right', code: 100 },
```

Comment:
I've inverted the expected object here, and have used `.code` as the accessor in tests. This helps us have one object and indexer for all tests